respa_exchange/downloader.py

```diff
@@ -319,19 +377,23 @@ def sync_from_exchange(ex_resource, future_days=365, no_op=False):
     }
 
     for item_id, item in calendar_items.items():
-        # Send the raw XML to Sentry for better debugging
-        with push_scope() as scope:
+        with configure_scope() as scope:
+            # Send the raw XML to Sentry for better debugging
             scope.set_extra('item_xml', element_to_string(item))
 
-            ex_reservation = extant_exchange_reservations.get(item_id.hash)
+        ex_reservation = extant_exchange_reservations.get(item_id.hash)
 
-            item_props = _parse_item_props(ex_resource, item_id, item)
+        item_props = _parse_item_props(ex_resource, item)
+
+        if not ex_reservation:  # It's a new one!
+            ex_reservation = _create_reservation_from_exchange(item_id, ex_resource, item_props)
+        else:
+            if ex_reservation._change_key != item_id.change_key:
+                # Things changed, so edit the reservation
+                _update_reservation_from_exchange(item_id, ex_reservation, ex_resource, item_props)
 
-            if not ex_reservation:  # It's a new one!
-                ex_reservation = _create_reservation_from_exchange(item_id, ex_resource, item_props)
-            else:
-                if ex_reservation._change_key != item_id.change_key:
-                    # Things changed, so edit the reservation
-                    _update_reservation_from_exchange(item_id, ex_reservation, ex_resource, item_props)
+    with configure_scope() as scope:
+        scope.remove_extra('item_xml')
+        scope.remove_extra('resource')
```

Comment:
It probably doesn't matter at all in practice, but when `no_op` is `True`, these won't be removed from the scope.